pkg/bpf/bpf_kmesh_common.go

```diff
@@ -94,26 +97,96 @@ func (sc *BpfSockConn) loadKmeshSockConnObjects() (*ebpf.CollectionSpec, error)
 	)
 	opts.Maps.PinPath = sc.Info.MapPath
 
-	spec, err = bpf2go.LoadKmeshCgroupSock()
+	if sc.Info.KernVerLt5_13 {
+		spec, err = bpf2go.LoadKmeshCgroupSockLo()
+	} else {
+		spec, err = bpf2go.LoadKmeshCgroupSock()
+	}
 
 	if err != nil || spec == nil {
 		return nil, err
 	}
 
 	SetInnerMap(spec)
 	setMapPinType(spec, ebpf.PinByName)
-	if err = spec.LoadAndAssign(&sc.KmeshCgroupSockObjects, &opts); err != nil {
+	if sc.Info.KernVerLt5_13 {
+		err = spec.LoadAndAssign(&sc.bpfObjLo, &opts)
+	} else {
+		spec.LoadAndAssign(&sc.bpfObj, &opts)
+	}
+	if err != nil {
 		return nil, err
 	}
 
-	value := reflect.ValueOf(sc.KmeshCgroupSockObjects.KmeshCgroupSockPrograms)
+	var value reflect.Value
+	if sc.Info.KernVerLt5_13 {
+		value = reflect.ValueOf(sc.bpfObjLo.KmeshCgroupSockLoPrograms)
+	} else {
+		value = reflect.ValueOf(sc.bpfObj.KmeshCgroupSockPrograms)
+	}
 	if err = pinPrograms(&value, sc.Info.BpfFsPath); err != nil {
 		return nil, err
 	}
 
 	return spec, nil
 }
 
+func (sc BpfSockConn) GetKmeshTailCallProgMap() *ebpf.Map {
+	if sc.Info.KernVerLt5_13 {
```

Comment:
It is not a good solution to add version judgment branches to the code.
This problem is caused by the fact that kmesh depends on some capabilities of the kernel, which are not supported by the compilation environment and running environment in the same way.
Is it better to solve this problem in a project, for example, to compile two Kmesh versions with different capabilities and select a proper kmesh based on the runtime environment?